app/models/split.rb

```diff
@@ -2,7 +2,7 @@ class Split < ActiveRecord::Base
   belongs_to :owner_app, required: true, class_name: "App", inverse_of: :splits
 
   has_many :previous_split_registries, dependent: :nullify
-  has_many :assignments, dependent: :nullify
+  has_many :assignments, -> { for_presentation }, dependent: :nullify, inverse_of: :split
```

Comment:
For reference, I've never encountered a good way to pass arguments through from associations to scopes, so the `.assignments` association will probably never be able to dynamically adjust to anything other than the current state of the DB. This might make it tricky in certain controllers where you need to do `assignments.for_presentation(some_args)`, because it'll already have `for_presentation` (no args) baked into it

Reply:
Yeah, we'll see how to do this. I might make a pseudo-association instead. Its usage is read-only anyway.

Reply:
I think the correct factoring for all this will become clearer as I need to wire it into the mobile-specific controllers and provide it an app_build.

Reply:
It might also be possible to assign transient state to the split instance and then read that from the association, but that seems like all kinds of magic.

Reply:
yep. i will try desperately to avoid. :)